modules/distributor/distributor.go

```diff
@@ -397,15 +405,54 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			return err
 		}
 
-		_, err = c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
 		metricIngesterAppends.WithLabelValues(ingester.Addr).Inc()
-		if err != nil {
+
+		if err != nil { // internal error, drop entire batch
 			metricIngesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
+			return err
 		}
-		return err
+
+		// assign a number for each index that had an error
+		// maxLive = 1, traceTooLarge = 2
+		// indexes that do not have an error will have a default 0 value
+		ringBatchResults := make([]int, len(indexes))
```

Comment:
i'm fine with something simpler. we don't need to perfectly count all reasons failed, right?  for instance in RF3 if a trace is rejected once for max live and once for too large then there is no "right answer" for how to metric dropped spans. all we really need to know is total errors as well as one (the first?) of the rejection reasons.

Reply:
allocating this for every push might get costly. can't we translate the response data directly into the batchResults from the response using the indexes?